src/criteria/default/articles-read.js

```diff
@@ -0,0 +1,29 @@
+import { setMatchingAttribute } from '../utils';
+
+setMatchingAttribute( 'articles_read', ras => {
+	const views = ras.getUniqueActivitiesBy( 'article_view', 'post_id' );
+	return views.filter( view => view.timestamp > Date.now() - 30 * 24 * 60 * 60 * 1000 ).length;
+} );
+
+setMatchingAttribute( 'articles_read_in_session', ras => {
+	const views = ras.getUniqueActivitiesBy( 'article_view', 'post_id' );
+	if ( ! views.length ) {
+		return 0;
+	}
+	// Sort by descending timestamp.
+	views.sort( ( a, b ) => b.timestamp - a.timestamp );
+	// Bail if the most recent view is older than 30 days.
```

Comment:
Wouldn't we want to bail if the most recent view is older than 30 _minutes_? If that's the case, then any views couldn't be considered the current session, right?

Reply:
Whoops, that was a typo in the inline comment. The code is 30 minutes. Fixed in 3bee99371212cca106d3555778087aa4569829ab